src/resolvers/resolver.ts

```diff
@@ -1,26 +1,64 @@
 import mongoose from "mongoose";
 import { Profile, User } from "../models/user";
+import bcrypt from "bcryptjs";
+import * as jwt from "jsonwebtoken";
+
+const SECRET = process.env.SECRET || "test_secret";
 
 const resolvers = {
   Query: {
-    hello: () => "Hellooo, welcome to your Graphql server",
-    async getAllUsers() {
-      const users = await User.find({});
-      return users;
-    },
-    async getAllProfiles() {
+    async getAllProfiles(_: any, _args: any, context: { userId: any }) {
+      if (!context.userId) throw new Error("Unauthorized");
       const profiles = await Profile.find({});
       return profiles;
     },
   },
   Mutation: {
-    async createUser(_: any, { email, password }: any) {
+    async createUser(
+      _: any,
+      { registerInput: { email, password, role } }: any
+    ) {
       const userExists = await User.findOne({ email: email });
       if (userExists) throw new Error("Email is taken");
-      const newUser = await User.create({ email, password });
-      return newUser;
+      const hashedPassword = await bcrypt.hash(password, 10);
+
+      const newUser = await User.create({
+        role: role || "user",
+        email: email,
+        password: hashedPassword,
+      });
+      const token = jwt.sign(
+        { userId: newUser._id, role: newUser?.role },
+        SECRET,
+        { expiresIn: "2h" }
+      );
+
+      return { token, user: newUser };
+    },
+    async loginUser(_: any, { loginInput: { email, password } }: any) {
```

Comment:
After creating a user and assigning him/her a role, like an admin role. the user is always authenticated as a "user" on login